troposphere/static/js/components/modals/image_version/ImageVersionEditModal.react.js

```diff
@@ -159,9 +167,23 @@ export default React.createClass({
     // Render
     // ------
     //
+    validateName: function(image, name) {
+        let orig_version = this.props.version,
+        versions = image.get('versions');
 
+        name = $.trim(name).toLowerCase();
+        versions = versions.filter(function(version) {
+            return (version.id !== orig_version.id && version.name.toLowerCase() === name);
+        });
+        return versions.length == 0;
+    },
     handleNameChange: function(name){
-      this.setState({versionName: name});
+      let image = this.state.versionImage;
+      let versionNameError = this.validateName(image, name) ? "" : "There already exists a version named '"+name+"' in application '"+image.get('name')+"'.";
```

Comment:
I think we could use a [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) might help the _else_ case for this ternary: 

```
let versionNameError = this.validateName(image, name) ? "" : 
    `There already exists a version named '${name}' in application '${image.get('name')}'.`;
```
example: [babel repl](https://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&experimental=false&loose=false&spec=true&code=%0Afunction%20foo()%20%7B%0A%20%20let%20name%20%3D%20%221.0.0%22%3B%0A%20%20let%20image%20%3D%20%7B%0A%20%20%20%20get%3A%20function%20(s)%20%7B%20return%20s%3B%20%7D%0A%20%20%7D%3B%0A%0A%20%20let%20versionNameError%20%3D%20this.validateName(image%2C%20name)%20%3F%20%22%22%20%3A%20%0A%20%20%20%20%60There%20already%20exists%20a%20version%20named%20'%24%7Bname%7D'%20in%20application%20'%24%7Bimage.get('name')%7D'.%60%3B%0A%0A%20%20console.log(%22foo!%22)%3B%0A%7D%0A%0Afunction%20validateName(image%2C%20name)%20%7B%0A%20%20return%20true%3B%0A%7D%0A%0Afunction%20qux()%20%7B%0A%20%20let%20name%20%3D%20%221.0.0%22%3B%0A%20%20let%20image%20%3D%20%7B%0A%20%20%20%20get%3A%20function%20(s)%20%7B%20return%20s%3B%20%7D%0A%20%20%7D%3B%0A%20%20let%20versionNameError%20%3D%20this.validateName(image%2C%20name)%20%3F%20%22%22%20%3A%20%0A%20%20%20%20%22There%20already%20exists%20a%20version%20named%20'%22%2Bname%2B%22'%20in%20application%20'%22%2Bimage.get('name')%2B%22'.%22%3B%0A%0A%20%20console.log(versionNameError)%3B%0A%0A%7D%0A%0Aexport%20%7B%20foo%2C%20qux%20%7D%3B%20%0A)

Could even consider doing an ExtractMethod refactor of this to a function, `getVersionNameError(image, name)` ...